refactor(explore): Enhance Dataset and Control panel Collapse components

geido opened the pull request:
### SUMMARY
Removes react-bootstrap Collapse with Antdesign enhanced Collapse.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:

https://user-images.githubusercontent.com/60598000/105727326-41f8f800-5f2b-11eb-8ae3-bf7ee4d7e866.mp4

After:

https://user-images.githubusercontent.com/60598000/105727342-46bdac00-5f2b-11eb-9826-dae94ae274dd.mp4

### TEST PLAN
1. Open the explorer
2. Make sure the control panel has all the collapsible expanded
3. Close a collapsible panel, the content should be hidden
4. Re-open the closed collapsible panel, the content should appear

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

geido committed:
Enhance control panel Collapse

geido commented:
@rusackas @junlincc

junlincc commented:
thanks @geido for the PR. if the caret was the only thing that changed, it LGTM. but im not sure why the tabs in the control panel got messed up a bit... 
<img width="257" alt="Screen Shot 2021-01-01 at 2 10 13 PM" src="https://user-images.githubusercontent.com/67837651/103447054-1d1aa900-4c3b-11eb-8443-9f8b98a44070.png">
this is what im seeing and how it supposed to look in another open PR
<img width="622" alt="Screen Shot 2021-01-01 at 2 11 47 PM" src="https://user-images.githubusercontent.com/67837651/103447104-9b774b00-4c3b-11eb-8bd0-3813d074781f.png">

geido commented:
> thanks @geido for the PR. if the caret was the only thing that changed, it LGTM. but im not sure why the tabs in the control panel got messed up a bit... 
> <img width="257" alt="Screen Shot 2021-01-01 at 2 10 13 PM" src="https://user-images.githubusercontent.com/67837651/103447054-1d1aa900-4c3b-11eb-8443-9f8b98a44070.png">
> this is what im seeing and how it supposed to look in another open PR
> 
> 
> <img width="622" alt="Screen Shot 2021-01-01 at 2 11 47 PM" src="https://user-images.githubusercontent.com/67837651/103447104-9b774b00-4c3b-11eb-8bd0-3813d074781f.png">
> 

Thanks @junlincc. Let me check what might have caused that.

geido committed:
Merge remote-tracking branch 'upstream/master' into control_panel_enhanced_collapse

geido committed:
Fix wrong import and minor refactor

geido commented:
@junlincc Please check now. The issue should be resolved.

junlincc reviewed (approved):
manually tested, LGTM!

If it's easy, please remove 'Dataset' from the Dataset & Chart Type Section in this PR as well. 
<img width="502" alt="Screen Shot 2021-01-04 at 9 07 55 PM" src="https://user-images.githubusercontent.com/67837651/103608791-01c8cb80-4ed1-11eb-888d-e5978bd9faa4.png">

and get the tests passed.

geido committed:
Fix for missing expanded param

geido committed:
Merge remote-tracking branch 'upstream/master' into control_panel_enhanced_collapse

ktmud commented:
The caret directions are inconsistent with the datasource panel.

**Folded:**
![image](https://user-images.githubusercontent.com/335541/103692666-f9ae7180-4f4c-11eb-80b7-28e621559e29.png)

**Expanded:**

![image](https://user-images.githubusercontent.com/335541/103692728-0d59d800-4f4d-11eb-9f2d-e5f926b75291.png)


I think the directions used by the datasource pane makes more sense.

ktmud commented:
Also, can we fix the font-size inconsistency in these panes as well (shouldn't they use the same styled component)?

<img src="https://user-images.githubusercontent.com/335541/103692824-34180e80-4f4d-11eb-92d0-f98e9a745343.png" width="300" />

ktmud commented:
May I also suggest tune the font size/spacing of the columns/metrics as well?

**Before**:
<img src="https://user-images.githubusercontent.com/335541/103693412-1a2afb80-4f4e-11eb-9d6d-1646d9f84aa6.png" width="500">

**After**:
<img src="https://user-images.githubusercontent.com/335541/103693357-fff11d80-4f4d-11eb-8537-71eba26a53ff.png" width="500">
(a rough adjustment, could be better)

ktmud commented:
<img src="https://user-images.githubusercontent.com/335541/103693644-7d1c9280-4f4e-11eb-8821-2d5161c613a9.png" width="500">

The title border spacing is also inconsistent.

geido commented:
Hello @ktmud. Thanks for your feedback. We are working on moving the Collapse components to Antd in small steps. This is the reason why you see inconsistencies between the two. We should use Antdesign as a reference though.

Concerning the fonts, react-bootstrap was having a 16px overriding the default 14px. This should be an improvement. Unless you think keeping the 16px imposed by react-bootstrap was intentional.

As for the direction of the arrows, those are by default in Antd. While I agree with you that the directions used by the previous Collapse are more intuitive, this is a change that will have an impact on all the Collapse components already moved to Antd. I would keep it as an item for later.

ktmud commented:
Cool. Glad to hear this is an ongoing migration. Just documenting what I found here in case they slip through.

geido commented:
After another look, I have noticed that the DatasourcePanel is using the Antd Collapse already, just not the enhanced one that we have introduced in the common components, but another flavor of it. This will need to be consolidated. Most probably all the existing instances of the Antd Collapse that aren't using the enhanced one will need to be moved to the enhanced.  Thoughts @rusackas?

geido committed:
Update cypress test

geido committed:
Merge remote-tracking branch 'upstream/master' into control_panel_enhanced_collapse

geido committed:
Merge remote-tracking branch 'upstream/master' into control_panel_enhanced_collapse

geido committed:
Fix merge conflicts

geido committed:
Apply consistent styling - in progress

rusackas commented:
After much discussion with the Preset design team (other voices can pitch in here!) we think we'd like to keep the current arrow directions. 

As far as arrow **direction** goes, the design proposal is this: If the collapse arrow is on the right side of the content area (like controls), it will point up when open or down when closed.. If it's on the left side of a header (like the Data pane) it will point right when closed, and down when open.

The above is more or less in accordance with the Apple [guidelines](https://developer.apple.com/design/human-interface-guidelines/macos/buttons/disclosure-controls/).

We should utilize a prop in the Collapse component to pick which version we're using... it appears the `expandIconPosition` could trigger conditional styling.

There's also a discrepancy in **animation** - we agreed that they should all be animated (and quickly!) with clockwise rotation when opening, and counterclockwise when closing.

And in coming to all these conclusions, we came up with a whole system of floating design proposals as Github Issues, then moving them to the Github wiki when consensus is reached. We'll be sending a proposal around this to the `dev@` list soon!

geido committed:
Align DataSource panel to enhanced component

ktmud commented:
While we are here, can we also fix the click area?

Currently the Dateset panel accordions can click to fold/expand on the whole header section, but the Controls panel accordions only have the text and the caret clickable.

geido committed:
Clean up DataSource panels style

ktmud commented:
Personally I'm also not a fan of caret animation (and even the slide up/slide down animation maybe we can also get rid of). 

Airbnb design guidelines for internal tools is also avoid animation as much as possible. We don't even have animation for modals and popovers.

For tools people will use everyday, (the presumed) efficiency and speed are probably more important than the "pleasant surprises" animations are normally designed to inspire.

geido committed:
Fix merge conflicts

geido commented:
> While we are here, can we also fix the click area?

@ktmud this PR will bring that for free as the Antdesign Collapse has the full header area clickable

geido commented:
@rusackas @ktmud I have made some adjustments so that the Dataset panel and Control panels use the same styling. As for the open points, concerning animation and direction of the arrows, let's come to a consensus so that I can close this up.

I tend to agree to avoid animations. As for the direction of the arrows, I am fine with what Evan has proposed.

geido committed:
Merge remote-tracking branch 'upstream/master' into control_panel_enhanced_collapse

geido committed:
Add optional animateArrows prop

geido committed:
Add storybook entry

geido commented:
I have made more changes as follows:

- Added an `animateArrows` prop so that those animations can be optional
- Implemented the arrow open/close position based on whether the `expandIconPosition` is right or left
- Added a storybook entry to show the arrows when the animation is ON
- I have enhanced also the DataSource panel, so it looks the same as the Control panel
- I have also updated the description with the updated videos showing the functionality

This is quite a big change. I will appreciate support with testing eventual edge cases.

Thank you!

@junlincc @rusackas @ktmud

maloun96 commented on superset-frontend/src/common/components/Collapse.tsx:
```diff
@@ -68,6 +90,18 @@ const Collapse = Object.assign(
+          expandIconPosition === 'right' &&
+          `
+            .anticon.anticon-right.ant-collapse-arrow > svg {
+              transform: rotate(-90deg) !important;
```
can we avoid using important?

geido replied:
Nope. This is added by the library on the element itself, which makes the important necessary in order to override

ktmud reviewed (approved):
LGTM. Hope we can have a `Collapse.test.tsx` file but that's a low priority.

adam-stasiak commented:
I see that when I click on second collapse then field with ```search metrics and columns``` jumps a little bit.
Are we able to fix this?

https://user-images.githubusercontent.com/25153919/105753898-ad51c280-5f49-11eb-8117-61a8d4b6fb7e.mov

geido commented:
Hello @adam-stasiak I am seeing the same problem on master. Would you like to create a separate issue for that?

https://www.awesomescreenshot.com/video/2508610?key=292778fadfc41d2f0f20f3099117aea6

Thanks

adam-stasiak commented:
ah ok! I will create! sorry for inconvenience.

geido committed:
Remove negative margin

adam-stasiak commented:
after manual testing looks fine to me 🟢
